### PR TITLE
Code gen fix

### PR DESCRIPTION
**Pull request by LouisSzeto**
#### Description
- Update script to include all Index/IndexOption timezone
- (Manual) fix double display C#/Py snippets https://www.quantconnect.com/docs/v2/writing-algorithms/securities/asset-classes/us-equity/corporate-fundamentals#02-Properties
- URL fixes according to URL testers and update URL testers to consider special case `Mixed-Mode Consolidator`.
- All code gen that erase previous version output to retain `metadata.json`
- Rerun code gen to generate correct content

#### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking change which improves implementation)
- [ ] New feature (non-breaking change which adds functionality)

#### Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: Resources/datasets/market-hours/CryptoFuture.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<h4>BINANCE</h4>
+<table class="table qc-table table-reflow">
+<thead>
+<tr><th>Symbol</th><th>Contract</th></tr>
+</thead>
+<tbody>
```
Comment (DerekMelchin) [inline Resources/datasets/market-hours/CryptoFuture.html:6]: Is this table supposed to be empty?

File: 02 Writing Algorithms/03 Securities/99 Asset Classes/07 Futures/04 Market Hours/00.json [hunk 2]
```diff
@@ -2,6 +2,6 @@
   "type" : "landing",
   "heading" : "Market Hours",
   "subHeading" : "",
-  "content" : "<p>Select one of the following Futures markets to see its operating hours:</p>",
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/03 Securities/99 Asset Classes/07 Futures/04 Market Hours/00.json:5]: I think we should keep this new heading

File: Resources/datasets/market-hours/cryptofuture/binance/generic/holidays.html [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<!-- Code generated by market-hour-code-generator.py -->
+<p>There is no holidays for the BINANCE CryptoFuture market.</p>
```
Comment (DerekMelchin) [inline Resources/datasets/market-hours/cryptofuture/binance/generic/holidays.html:2]: Change "BINANCE" to "Binance"

File: Resources/datasets/market-hours/index/usa/generic/time-zone.html [hunk 4]
```diff
@@ -1,3 +1,3 @@
 <!-- Code generated by market-hour-code-generator.py -->
 
-<p>The US Index market trades in the <code>America/Chicago</code> time zone.</p>
\ No newline at end of file
+<p>The US Index market trades in the <code>America/Chicago, America/New York</code> time zone.</p>
```
Comment (DerekMelchin) [inline Resources/datasets/market-hours/index/usa/generic/time-zone.html:3]: These should be separated by an "and". If there are more than two, maybe we should use a list

> The US Index market trades in the following time zones:
> - <code>America/Chicago</code>
> - <code>America/New York</code>
> - <code>America/New York</code>

File: Resources/enumerations/market-index.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<p>The following <code>Market</code> enumeration members are available for Index:</p>
```
Comment (DerekMelchin) [inline Resources/enumerations/market-index.html:1]: Let's keep these plural. "Index" => "Indices"

File: Resources/enumerations/market-option.html [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<p>The following <code>Market</code> enumeration members are available for Option:</p>
```
Comment (DerekMelchin) [inline Resources/enumerations/market-option.html:1]: "Options"

**Review by DerekMelchin — CHANGES_REQUESTED** [review]

File: Resources/datasets/market-hours/index/usa/generic/time-zone.html (hunk 4 above)
Comment (DerekMelchin) [reply Resources/datasets/market-hours/index/usa/generic/time-zone.html:3]: If there are multiple time zones, "time zone" needs to be "time zones".

File: 02 Writing Algorithms/03 Securities/99 Asset Classes/07 Futures/04 Market Hours/00.json (hunk 2 above)
Comment (LouisSzeto) [reply 02 Writing Algorithms/03 Securities/99 Asset Classes/07 Futures/04 Market Hours/00.json:5]: np. Should be inherited from an old version only

File: Resources/datasets/market-hours/CryptoFuture.html (hunk 1 above)
Comment (LouisSzeto) [reply Resources/datasets/market-hours/CryptoFuture.html:6]: Yes. The converter will handle that when converting to HTML page

File: Resources/datasets/market-hours/index/usa/generic/time-zone.html (hunk 4 above)
Comment (LouisSzeto) [reply Resources/datasets/market-hours/index/usa/generic/time-zone.html:3]: How about we use a list for all cases? It looks easier to follow to me.

File: 02 Writing Algorithms/03 Securities/99 Asset Classes/07 Futures/04 Market Hours/11 CBOT/08 Time Zone.html [hunk 7]
```diff
@@ -1,3 +1,6 @@
 <!-- Code generated by market-hour-code-generator.py -->
 
-<p>The CBOT Future market trades in the <code>America/Chicago</code> time zone.</p>
\ No newline at end of file
+<p>The CBOT Future market trades in the following time zone(s).</p>
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/03 Securities/99 Asset Classes/07 Futures/04 Market Hours/11 CBOT/08 Time Zone.html:3]: We should only use a list if there multiple items.

If there is only 1 time zone, we should use 
> The CBOT Future market trades in the <code>America/Chicago</code> time zone.

If there is multiple time zones, we should use 
> The CBOT Future market trades in the following time zones:

and then put a list.

Note that `.` is now `:`

File: Resources/enumerations/market-indexoption.html [hunk 8]
```diff
@@ -1,3 +1,3 @@
-<p>The following <code>Market</code> enumeration members are available for Index Options:</p>
+<p>The following <code>Market</code> enumeration members are available for Indices Options:</p>
```
Comment (DerekMelchin) [inline Resources/enumerations/market-indexoption.html:1]: "Indices" is good in most cases, but in the case of "Indices Options", it should really be "Index Options"

**Review by DerekMelchin — CHANGES_REQUESTED** [review]

**Review by DerekMelchin — APPROVED** [review]

**Pull request merged**